web/src/widgetize.ts

```diff
@@ -1,17 +1,45 @@
 import $ from "jquery";
+import assert from "minimalistic-assert";
 
 import * as blueslip from "./blueslip";
 import * as message_lists from "./message_lists";
+import type {Message} from "./message_store";
 import * as narrow_state from "./narrow_state";
-
-export const widgets = new Map();
-export const widget_contents = new Map();
-
-export function clear_for_testing() {
+import type {Event} from "./poll_widget";
+
+type WidgetOptions = {
+    widget_type: string;
+    extra_data:
+        | {question: string; options: string[]}
+        | {
+              type: string;
+              heading: string;
+              choices: {type: string; reply: string; long_name: string; short_name: string}[];
+          }
+        | undefined;
+    events: Event[];
+    $row: JQuery;
+    message: Message;
+    post_to_server: (data: {msg_type: string; data: string}) => void;
+};
+
+type WidgetValue = Record<string, unknown> & {
+    activate: (data: {
+        $elem: JQuery;
+        callback: (data: string) => void;
+        message: WidgetOptions["message"];
```

Comment:
Is there a difference b/w `WidgetOptions["message"]` and mere `Message` type. I think Message would be a bit more clear.

Reply:
yah correct! Idk why I had written it like this. will fix.